src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java

```diff
@@ -6,154 +6,181 @@
+                             @NotNull CloudStackUser cloudStackUser) throws FogbowException {
+
+        URIBuilder uriRequest = request.getUriBuilder();
+        CloudStackUrlUtil.sign(uriRequest, cloudStackUser.getToken());
+
+        try {
+            String jsonResponse = this.client.doGetRequest(uriRequest.toString(), cloudStackUser);
+            CreateFirewallRuleAsyncResponse response = CreateFirewallRuleAsyncResponse.fromJson(jsonResponse);
+            try {
+                return CloudStackCloudUtils.waitForResult(
+                        this.client, this.cloudStackUrl, response.getJobId(), cloudStackUser);
+            } catch (CloudStackCloudUtils.TimeoutCloudstackAsync e) {
+                CloudStackQueryAsyncJobResponse asyncJobResponse = CloudStackCloudUtils.getAsyncJobResponse(client,
+                        cloudStackUrl, response.getJobId(), cloudStackUser);
+                deleteSecurityRule(asyncJobResponse.getJobInstanceId(), cloudStackUser);
+                throw e;
             }
-            throw new FogbowException(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
-        } else {
-            throw new UnexpectedException();
+        } catch (HttpResponseException e) {
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
         }
     }
 
-	protected List<SecurityRuleInstance> getFirewallRules(String ipAddressId, CloudStackUser cloudUser) throws FogbowException {
+    @NotNull
+    @VisibleForTesting
+	List<SecurityRuleInstance> getFirewallRules(String ipAddressId, @NotNull CloudStackUser cloudUser)
+            throws FogbowException {
+
 		ListFirewallRulesRequest request = new ListFirewallRulesRequest.Builder()
 				.ipAddressId(ipAddressId)
 				.build(this.cloudStackUrl);
 
-		CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
+        URIBuilder uriRequest = request.getUriBuilder();
+        CloudStackUrlUtil.sign(uriRequest, cloudUser.getToken());
 
-		String jsonResponse = null;
 		try {
-			jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
-		} catch (HttpResponseException e) {
-			CloudStackHttpToFogbowExceptionMapper.map(e);
-		}
-
-		ListFirewallRulesResponse response = ListFirewallRulesResponse.fromJson(jsonResponse);
-		List<ListFirewallRulesResponse.SecurityRuleResponse> securityRulesResponse = response.getSecurityRulesResponse();
-		return convertToFogbowSecurityRules(securityRulesResponse);
+			String jsonResponse = this.client.doGetRequest(uriRequest.toString(), cloudUser);
+            ListFirewallRulesResponse response = ListFirewallRulesResponse.fromJson(jsonResponse);
+            List<ListFirewallRulesResponse.SecurityRuleResponse> securityRulesResponse =
+                    response.getSecurityRulesResponse();
+            return convertToFogbowSecurityRules(securityRulesResponse);
+        } catch (HttpResponseException e) {
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
+        }
```

Comment:
Fix the indentation of this method.

Reply:
I've fixed this in the PR five.

Reply:
My IDE wasn't catching this indentation mistake...
Has anyone ever had this problem before?

Reply:
Apparently IDEs don't show these indentation errors, but you can set them not to use tab spaces and replace these tabs with 4 spaces.